File: leetcode/242. Valid Anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size() != t.size()){
            return false;
        }
        unordered_map<char, int> m;
        for(int i = 0; i < s.size(); ++i){
            auto iter = m.find(s[i]);
            if(iter != m.end()){
                ++m[s[i]];
            } else {
                m.emplace(s[i], 1);
            }
        }
        
        for(int i = 0; i < t.size(); ++i){
            auto iter = m.find(t[i]);
            if(iter == m.end()){
                return false;
            } else if(m[t[i]] == 0) {
                return false;
            } else {
                --m[t[i]];
            }
        }
        return true;
    }
};

File: leetcode/242. Valid Anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) != len(t):
            return False
        dic = [0 for i in range(26)]
        for c in s:
            dic[ord(c)-ord('a')] += 1
        for c in t:
            if dic[ord(c)-ord('a')] > 0:
                dic[ord(c)-ord('a')] -= 1
            else:
                return False
        return True
